clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -374,7 +406,9 @@ private void handleListOffsetResponse(TopicPartition topicPartition,
                 }
 
                 long offset = this.subscriptions.fetched(partition);
-                fetch.put(partition, new FetchRequest.PartitionData(offset, this.fetchSize));
+                long consumed = this.subscriptions.consumed(partition);
+                if (consumed == offset)
```

Comment:
Might be worth a comment explaining why this check is needed. Basically it ensures that Fetcher will have no more than one outstanding request for any partition, which prevents fetches from getting too far ahead. Is that right? You might also mention that it doesn't impact prefetching since the consumed position will always be updated prior to initializing the next fetches.

Also, rename 'offset' to 'fetched'?